test/rubygems/test_gem_version.rb

```diff
@@ -226,6 +226,65 @@ def test_frozen_version
     assert_version_equal v("2"), v.bump
   end
 
+  def test_deconstruct
+    version = v("3.2.1")
+    major, minor, patch = version.deconstruct
+    assert_equal 3, major
+    assert_equal 2, minor
+    assert_equal 1, patch
+  end
+
+  def test_deconstruct_keys
+    version = v("3.2.1")
+    assert_equal({ major: 3, minor: 2, build: 1 }, version.deconstruct_keys(nil))
+  end
+
+  def test_deconstruct_keys_two_segments
+    version = v("3.2")
+    assert_equal({ major: 3, minor: 2, build: nil }, version.deconstruct_keys(nil))
+  end
+
+  def test_deconstruct_keys_one_segment
+    version = v("3")
+    assert_equal({ major: 3, minor: nil, build: nil }, version.deconstruct_keys(nil))
+  end
+
+  def test_pattern_matching_array
+    case v("3.2.1")
+    in [major, minor, build]
+      assert_equal 3, major
+      assert_equal 2, minor
+      assert_equal 1, build
+    else
+      flunk "Array pattern did not match"
+    end
+  end
+
+  def test_pattern_matching_hash
+    result =
+      case v("3.2.1")
+      in major: 3.., minor: 2.. then "matched"
+      else "no match"
+      end
+    assert_equal "matched", result
+  end
+
+  def test_pattern_matching_hash_vs_comparison
+    # Hash pattern checks each segment independently
+    version = v("4.0")
+    result =
+      case version
+      in major: (3..), minor: (2..)
```

Comment:
See above comment on parens. TL;DR: without them it tries to do `2.."matches"` because of the newline.